astropy/table/operations.py

```diff
@@ -697,6 +697,69 @@ def common_dtype(cols):
         raise tme
 
 
+def _get_join_sort_idxs(out_descrs, keys, len_left, len_right, left, right):
+
+    # Remake out_descrs as a dict keyed by the name (descr[0])
+    descrs = {descr[0]: descr for descr in out_descrs}
+
+    # Go through each of the key columns in order and make columns for
+    # a new structured array that represents the lexical ordering of those
+    # key columns. This structured array is then argsort'ed. The trick here
+    # is that some columns (e.g. Time) may need to be expanded into multiple
+    # columns for ordering here.
+
+    ii = 0  # Index for uniquely naming the sort columns
+    sort_keys_dtypes = []  # sortable_table dtypes as list of (name, dtype_str, shape) tuples
+    sort_keys = []  # sortable_table (structured ndarray) column names
+    sort_left = {}  # sortable ndarrays from left table
+    sort_right = {}  # sortable ndarray from right table
+
+    for key in keys:
+        # TODO: shape here seems out of place. Can a key column have a non-trivial shape?
```

Comment:
This `shape` is not actually used now.

Reply:
I wondered about this - it seems it should be impossible to sort on anything but 1-dimensional columns.